modules/packages/LinearAlgebra.chpl

```diff
@@ -2500,6 +2517,410 @@ proc cosm(A: []) throws {
+  var A: [Dom] eltType = 1 : eltType;
+  return A;
+}
+
+
+// Given the Domain and element-type, method
+// returns a Matrix of ones.
+private proc zeros(Dom: domain(2), type eltType=real) {
+  var A: [Dom] eltType = 0 : eltType;
+  return A;
+}
+
+
+// Returns an index's unit vector of size n.
+private proc elementaryVector(n,ind) {
+  var v: [0..<n] real;
+  v[ind] = 1;
+  return v;
+}
+
+record Cmp {
+  param desc;
+  var A;
+}
+
+proc Cmp.compare(a, b) {
+  if desc then {
+    return -(A[a] - A[b]);
+  }
+  else {
+    return A[a] - A[b];
+  }
+}
+
+// Given an array A, returns indexes
+// corresponding to the sorted ordering of A.
+private proc argsort(A: [], param desc=false) {
+  use Sort;
+  var cmp = new Cmp(desc, A);
+  var B = [ii in A.domain] ii;
+  sort(B, comparator=cmp);
+  return B;
+}
+
+// Method returns false if there is atleast one
+// column in X which isn't parallel to a column
+// in Y. Else the method returns true.
+private proc everyColOfXParallelToColOfY(X: [?xD], Y: [?yD]){
+  var n = xD.shape(0);
+  for i in xD.dim(1) {
+    var b = X[.., i];
+    var fg = false;
+    for j in yD.dim(1) {
+      if && reduce (Y[.., j]==b) {
+        fg = true;
+        break;
+      }
+    }
+    if !fg {
+      return false;
+    }
+  }
+  return true;
+}
+
+// Rounds up the elements of the Matrix to
+// 1.0 if element >= 0
+// -1.0 if element < 0
+private proc signRoundUp(A: []){
+  forall a in A {
+    if a == 0.0 then {
+      a += 1.0;
+    }
+    a /= abs(a);
+  }
+  return A;
+}
+
+// Method returns true if there is atleast one
+// column which is parallel to one of its previous
+// columns. If Y is not None, then method also
+// checks on columns of Y.
+private proc columnNeedsResampling(A: [?D], col=0, Y: [], shouldResampleBasedOnY=false) {
+  var n = D.shape(0);
+  var b = A[.., col];
+  for i in 0..<col {
+    if && reduce (A[.., i]==b) {
+      return true;
+    }
+  }
+  if shouldResampleBasedOnY {
+    var yCol = Y.domain.shape(1);
+    for i in 0..<yCol {
+      if && reduce (Y[.., i]==b) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+private proc columnResample(A: [?D], col = 0) {
+  use Random;
+
+  fillRandom(A[.., col]);
+  A[.., col] = round(A[.., col])*2 - 1;
+  return A;
+}
+
+// Method returns the product of a Sparse and a Dense Matrix.
+private proc sparseDenseMatmul(A: [], B: []) where isSparseArr(A) || !isSparseArr(B) {
+  var resDom = {0..<A.domain.shape(0), 0..<B.domain.shape(1)};
+  var C: [resDom] A.eltType;
+
+  for i in 0..<B.domain.shape(1) {
+    C[.., i] = dot(A, B[.., i]);
+  }
+  return C;
+}
+
+// Given Sparse Matrix, Function returns the Dense Matrix.
+private proc getDense(A: []) where isSparseArr(A) {
+  const (m,n) = A.domain.shape;
+  var resDom = {0..<m, 0..<n};
+  var C: [resDom] A.eltType;
+
+  forall (i,j) in A.domain do C[i,j] = A[i,j];
+
+  return C;
+}
+
+proc onenormest(A: [?D], param t=2,param itmax=5) throws {
```

Comment:
Add documentation.

Also, I can't remember whether there was any discussion around this interface. At the very least, I prefer `oneNormEst` and `itMax`.